libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/building/BuildingProcessor.kt

```diff
@@ -48,18 +47,25 @@ internal object BuildingProcessor {
     fun queryBuildingOnWaypoint(
         action: BuildingAction.QueryBuildingOnWaypoint
     ): BuildingResult.GetDestination {
-        val waypoints = action.progress.navigationRoute.internalWaypoints()
-        val waypointIndex = action.progress.currentLegProgress?.legIndex!! + 1
-        val waypoint = waypoints.filter { it.isLegWaypoint() }.getOrNull(waypointIndex)
-        return BuildingResult.GetDestination(waypoint?.target ?: waypoint?.location)
+        val routeOptions = action.progress.route.routeOptions()
```

Comment:
the intention of doing this changed is EV waypoints, that added implicitly and this logic is not working.

Reply:
What exactly is not working? I don't see anything here that would be affected by adding EV waypoints.

Reply:
@RingerJK could you demonstrate the issue with a unit test? 

It seems we're trying to get this into the 2.10.0-rc.1, and it'd be nice to release that soon

Reply:
let me provide the sample:
- the SDK requested an EV route with 2 waypoints (origin and destination);
- received the route with an additional waypoint in the middle (EV waypoint);
- so we have the route with 3 waypoints, 2 legs, but originally we requested 1 leg route (2 waypoints) 
- as soon as we start to work with requested waypoints (RouteOptions) we get inconsistency because `routeProgerss#legIndex == 1` is actually **not** the destination, it's the **EV** waypoint.

Reply:
We have 2 APIs: `queryBuildingOnWaypoint` and `queryBuildingOnFinalDestination`. In the example provided above, I would guess that `queryBuildingOnWaypoint` would be triggered when the user reaches the 1st waypoint and `queryBuildingOnFinalDestination` will be triggered when they reach the final destination. What am I missing?

cc @RingerJK

Reply:
Oh, legIndex is also affected.
Then we can't merge it. Thanks, I'll continue investigating tomorrow.

Reply:
@abhishek1508 at my example when you arrive at the first waypoint and try to take waypoint data via `queryBuildingOnWaypoint` you will receive data for `destination` because the index of destination in `RouteOptions` is actually `1`. Does it make sense?